### PR TITLE
checkbox + radio: examples with icon (close #351)

### LINKED ISSUE #351: radio: add test (example) for radio-option with icon

### PR DESCRIPTION
**Pull request by mishanga**
Close #351 

Radio:

Checkbox:

Normal:

### CODE REVIEW

**Comment by mishanga** [comment]
@dfilatov take a look

File: common.blocks/button/button.deps.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
     mustDeps : { block : 'i-bem', elems : ['dom'] },
     shouldDeps : [
         {
-            elems : ['text'],
+            elems : ['icon', 'text'],
```
Comment (tadatuta) [inline common.blocks/button/button.deps.js:5]: это разве не опциональная зависимость?

Comment (mishanga) [inline common.blocks/button/button.deps.js:5]: Сейчас по ней приезжает только шаблоны про тег `i` для иконки, больше ничего. Стили уже лежат внутри темы.
Непонятно, как сделать лучше.

File: common.blocks/radio/radio.tests/simple.bemjson.js [hunk 2]
```diff
@@ -77,6 +77,24 @@
...
+            options : [
+                {
+                    val : 'vk',
+                    text : 'VK',
+                    icon : { elem : 'icon' }
+                },
+                {
+                    val : 'twitter',
+                    text : 'Twitter',
+                    icon : { elem : 'icon' }
```
Comment (dfilatov) [inline common.blocks/radio/radio.tests/simple.bemjson.js:93]: Is `{ elem : 'icon' }` here is elem of `button` block in fact?

Comment (mishanga) [inline common.blocks/radio/radio.tests/simple.bemjson.js:93]: Да, шаблоны применяются [в контексте button](https://github.com/bem/bem-components/blob/v2/common.blocks/radio-option/_type/radio-option_type_button.bemhtml#L38).

Comment (dfilatov) [inline common.blocks/radio/radio.tests/simple.bemjson.js:93]: It's not very clear (
Could you remind me, why we don't have separate block for icons?

Comment (mishanga) [inline common.blocks/radio/radio.tests/simple.bemjson.js:93]: It is not very clear, which solution to choose. I'll keep it in my mind :) We need more examples of using for making a decision.
For now I have such concept:
- `icon` block (`<i>`) — for decorating only, as an element of other blocks, not for independent using;
- `image` block (``) — for images, used as a content (logo, for example).

Comment (mishanga) [inline common.blocks/radio/radio.tests/simple.bemjson.js:93]: https://github.com/bem/bem-components/issues/384

**Pull request merged**